magma/generator.py

```diff
@@ -67,14 +67,22 @@ def _make_type(cls, *args, **kwargs):
     dct = DefineCircuitKind.__prepare__(name, bases)
     cls.__init__(dummy, *args, **kwargs)
     dct.update(dict(dummy.__dict__))
-    return DefineCircuitKind.__new__(cls, name, bases, dct)
+    # Use circuit bind for the instance,
+    # rather than inherited Generator2 bind
+    dct["bind"] = classmethod(DefineCircuitKind.bind)
+    circuit = DefineCircuitKind.__new__(cls, name, bases, dct)
```

Comment:
This we can do by making `bind` a member function on `Generator2` right?

Reply:
Can you clarify? The issue is that Generator2 defines a bind method, so that is used instead of looking up the Circuit base class method.

Reply:
Like what if we had

```python
class Generator2:
    def bind(self, ...):  # now self is the circuit
        DefineCircuitKind.bind(self, ...)
```

Actually why does the MRO use the `Generator2.bind` instead of `DefineCircuitKind.bind` anyway?

Reply:
I think because we use `cls` and have Circuit as a base (see here https://github.com/phanrahan/magma/pull/945/files#diff-8194325e1fce50138ceca2e91aaadbd8eac22e64573df83388bc066fdbcee985R66), so the default is to lookup in `cls` first, before looking at the bases.

Reply:
That doesn't work because `bind` is already a classmethod in Generator2 (used for binding a monitor generator to the generator), so we can't have both a classmethod and a member method with the same name.  I tried moving the classmethod to the metaclass but that also doesn't work, since it will use the member method.

Reply:
Ah yeah ok. Seems like a pretty interesting use case.